feat: 알림 서비스 전체 기능 구현 및 웹소켓 연결 상태 관리 추가

hdg5639 opened the pull request:
<!-- 
PR 제목 작성 가이드:
형식: [타입]: 간단한 설명 (#이슈번호)

타입 예시:
- feat: 새로운 기능 추가
- fix: 버그 수정  
- docs: 문서 수정
- style: 코드 포맷팅
- refactor: 코드 리팩토링
- test: 테스트 추가/수정
- chore: 빌드, 패키지 등 기타 작업

예시: feat: 사용자 로그인 API 구현 (#123)
-->

## 📝 작업 내용
<!-- 이번 PR에서 구현한 기능이나 수정한 내용을 구체적으로 작성해주세요 -->
- 사용하지 않는 이전 Notification Entity와 DTO 모두 삭제
- NotificationServiceImpl 주요 기능 구현  
    - 식단 알림, 식사 알림, 결제 알림, 리뷰 알림 전송 기능 구현
    - 알림 읽음 처리 기능 (단건 / 전체) 구현
    - 사용자별 안읽은 알림 개수 조회 기능 추가
- NotificationController REST API 구성  
    - 알림 전체 조회 API
    - 알림 읽음 처리 API
    - 모든 알림 읽음 처리 API
    - 안읽은 알림 개수 조회 API
- ChannelInterceptor + SessionManageService 적용으로 웹소켓 연결 상태 확인 기능 구현
- CustomWebSocketHandlerDecoratorFactory 개선  
    - afterConnectionClosed 시 ChannelInterceptor 세션 정보 정상 반영
- WebSocketConfig 개선  
    - DecoratorFactory 등록 방식 수정하여 ChannelInterceptor 주입 가능하도록 변경

## 🔗 관련 이슈
<!-- 관련된 이슈가 있다면 링크해주세요 -->
- Related to #50

## 💬 추가 요청사항
<!-- 리뷰어에게 특별히 확인받고 싶은 부분이나 질문이 있다면 작성해주세요 --> 
- 1차 코드라 프론트와 테스트하여 정상 동작하는지 확인 필요
- DB에서 DietNoficiation, ExternalDietInfo, ExternalPaymentInfo,  PaymentNoficiation, SurveyNofitication 테이블 삭제 부탁드립니다

## ✅ 체크리스트
### 코드 품질
- [x] 커밋 컨벤션 준수 (feat/fix/docs/refactor 등)
- [x] 불필요한 코드/주석 제거

### 테스트
- [x] 로컬 환경에서 동작 확인 완료
- [x] 기존 기능에 영향 없음 확인

### 리뷰
- [x] 적절한 리뷰어 지정 완료
- [x] 리뷰어 1명 이상 승인

### 배포 준비
- [x] 환경변수 추가/변경사항 문서화
- [x] DB 마이그레이션 필요 여부 확인
- [x] 배포 시 주의사항 없음

hdg5639 committed:
feat: 알림 서비스 전체 기능 구현 및 웹소켓 연결 상태 관리 추가

- NotificationServiceImpl 주요 기능 구현 (식단, 결제, 리뷰 알림 전송)
- 알림 읽음 처리 기능 추가 (단건/전체)
- 사용자별 안읽은 알림 개수 조회 기능 추가
- ChannelInterceptor + SessionManageService 적용으로 웹소켓 연결 상태 확인
- CustomWebSocketHandlerDecoratorFactory 이용하여 disconnect 처리 시 세션 정보 동기화
- NotificationController REST API 구성 (조회, 읽음 처리, unread count 조회)

Issue: #50

2-jjong commented on Backend/src/main/java/com/demo/nimn/dao/notification/NotificationDAO.java:
이슈 #40 을 보면 DAO Layer 제거하고 Repository로 통합하는 이슈가 있습니다. 잘 확인하시고 반영 바랍니다.

2-jjong commented on Backend/src/main/java/com/demo/nimn/controller/notification/NotificationController.java:
현재 PatchMapping을 사용하고 있는데 Cors allow 설정에서 Patch 메소드가 빠져 있어서 정상 동작하지 않을 것으로 생각 됩니다. @ryusemin 와 상의하여 수정 바랍니다.

2-jjong commented on Backend/src/main/java/com/demo/nimn/dto/notification/response/ResponseDTO.java:
용도 알 수 있게 클래스명 명확히 작성해주세요.

2-jjong commented on Backend/src/main/java/com/demo/nimn/websocket/ChannelInterceptor.java:
```diff
@@ -0,0 +1,55 @@
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class ChannelInterceptor implements org.springframework.messaging.support.ChannelInterceptor {
```
ChannelInterceptor import하고 사용하는 것으로 변경해주세요.

2-jjong reviewed (changes requested):
전체적으로 로깅 처리를 통일 할 필요가 있을 것 같습니다. Slf4j의 log를 사용하는 부분도 있고 println을 사용하는 부분도 있는데 log를 사용하는 것으로 통일해주세요.

추가로 PR 내용 중에 잘못 작성된 부분이 있으니 확인하고 지워주시고 **체크리스트 확인** 후 체크 바랍니다.
![image](https://github.com/user-attachments/assets/e1719a5d-898e-4d06-a606-89ff2269b316)

ryusemin reviewed (commented):
확인하였고 CORS 허용 METHOD에 PATCH도 추가하겠습니다

hdg5639 committed:
refactor: 알림 서비스 DAO 클래스 삭제 및 로그 통일

- 알림 서비스 DAO 클래스 삭제
- NotificationService에 Repository 직접 주입
- ChannelInterceptor 클래스명을 CustomChannelInterceptor로 변경
- CustomChannelInterceptor 로그를 println 에서 log로 변경

Issue: #50
Related: #40

hdg5639 committed:
refactor: DTO 클래스명을 용도에 맞게 명확하게 변경

- ResponseDTO → NotificationCountDTO

Issue: #50

2-jjong reviewed (approved):
잘 작성되었네요. 프론트랑 테스트 진행하시면 될 것 같습니다.
